openwisp_monitoring/check/tests/test_ping.py

```diff
@@ -109,8 +109,8 @@ def test_device_deleted(self):
         check.save()
         # dev = d
         # import ipdb; ipdb.set_trace()
-        d.delete()
         check = Check.objects.get(pk=check.pk)
+        d.delete()
```

Comment:
so there was already a test here? Why was it passing, do you have any idea?

Reply:
I am unsure but it looks like `perform_check` triggered the `check`'s deletion though on backtracing I didn't find anything to trigger check's deletion besides `content_type` :smiley:

Reply:
this is weird. I'll investigate myself once the rest of the PR is ready. Thanks